src/calvaDebug.ts

```diff
@@ -0,0 +1,295 @@
+    }
+
+    private _createVariableFromLocal(local: any[]): Variable {
+        return {
+            name: local[0],
+            value: local[1],
+            // DEBUG TODO: May need to check type of value. If it's a map or collection, we may need to set variablesReference to something > 0.
+            /** If variablesReference is > 0, the variable is structured and its children can be retrieved by passing variablesReference to the VariablesRequest. */
+            variablesReference: 0
+        }
+    }
+
+    protected async variablesRequest(response: DebugProtocol.VariablesResponse, args: DebugProtocol.VariablesArguments, request?: DebugProtocol.Request): Promise<void> {
+
+        const debugResponse = state.deref().get(DEBUG_RESPONSE_KEY);
+
+        response.body = {
+            variables: debugResponse.locals.map(this._createVariableFromLocal)
+        };
+
+        this.sendResponse(response);
+
+        vscode.window.showTextDocument(vscode.window.activeTextEditor.document);
+    }
+
+    protected async continueRequest(response: DebugProtocol.ContinueResponse, args: DebugProtocol.ContinueArguments, request?: DebugProtocol.Request): Promise<void> {
+
+        const cljSession = util.getSession('clj');
+        const { id, key } = state.deref().get(DEBUG_RESPONSE_KEY);
+
+        if (cljSession) {
+            cljSession.sendDebugInput(':continue', id, key).then(response => {
+                this.sendEvent(new StoppedEvent('breakpoint', CalvaDebugSession.THREAD_ID));
+            });
+        }
+
+        this.sendResponse(response);
+    }
+
+    protected async disconnectRequest(response: DebugProtocol.DisconnectResponse, args: DebugProtocol.DisconnectArguments, request?: DebugProtocol.Request): Promise<void> {
+
+        const cljSession = util.getSession('clj');
+        const { id, key } = state.deref().get(DEBUG_RESPONSE_KEY);
+
+        if (cljSession) {
+            cljSession.sendDebugInput(':quit', id, key);
+        }
+
+        const cljReplWindow = replWindows['clj'];
+
+        if (cljReplWindow) {
+            cljReplWindow.stopDebugMode();
+        }
+
+        this.sendResponse(response);
+    }
+
+    protected terminateRequest(response: DebugProtocol.TerminateResponse, args: DebugProtocol.TerminateArguments, request?: DebugProtocol.Request): void {
+
+        this.sendResponse(response);
+    }
+
+    protected customRequest(command: string, response: DebugProtocol.Response, args: any, request?: DebugProtocol.Request): void {
+
+        switch (command) {
+            case REQUESTS.SEND_TERMINATED_EVENT: {
+                this.sendEvent(new TerminatedEvent());
+                break;
+            }
+            case REQUESTS.SEND_STOPPED_EVENT: {
+                this.sendEvent(new StoppedEvent(args.reason, CalvaDebugSession.THREAD_ID, args.exceptionText));
+                break;
+            }
+        }
+
+        this.sendResponse(response);
+    }
+}
+
+CalvaDebugSession.run(CalvaDebugSession);
```

Comment:
Should this be here? (Not saying that it shouldn't, just wondering 😄 )

Reply:
Yeah, I don't know exactly what this does, to be honest, but it's in the mock debug example/template project.

```
 import { MockDebugSession } from './mockDebug';

 MockDebugSession.run(MockDebugSession);
```